### PR TITLE
Update golangci-lint to version 1.31.0

### PR DESCRIPTION
**Pull request by kolaente**
This PR updates golangci-lint to the latest version 1.31.0.

The upgrade introduced a new check for which I've fixed or disabled most cases.

### CODE REVIEW

File: modules/repository/check.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func gatherMissingRepoRecords(ctx context.Context) ([]*models.Repository, error)
 			return nil
 		},
 	); err != nil {
-		if strings.HasPrefix("Aborted gathering missing repo", err.Error()) {
+		if strings.HasPrefix(err.Error(), "Aborted gathering missing repo") {
```
Comment (kolaente) [inline modules/repository/check.go:123]: Pinging @zeripath since I don't know if that was an intentional change in #10745 (it was reported by gocritic)

Comment (6543) [reply modules/repository/check.go:123]: for me this looks like a bug ...
Definiton:
```go
func HasPrefix(s, prefix string) bool
```

**Review by 6543 — APPROVED** [review]

File: modules/repository/check.go (hunk 1 above)
Comment (zeripath) [reply modules/repository/check.go:123]: I think that was a bug I must have added! Good catch!

**Review by zeripath — APPROVED** [review]
FYI ReplaceAll is golang version >1.12 - which is why it wasn't previously used.

**Comment by 6543** [comment]
:rocket:

**Comment by kolaente** [comment]
> FYI ReplaceAll is golang version >1.12 - which is why it wasn't previously used.

Ah makes sense, thanks for noting.

**Pull request merged**